packages/x-charts/src/context/CartesianProvider/zoom.ts

```diff
@@ -0,0 +1,95 @@
+import { AxisConfig } from '../../models';
+import { ExtremumGettersConfig } from '../PluginProvider';
+import { FormattedSeries } from '../SeriesProvider';
+import { getAxisExtremum } from './getAxisExtremum';
+
+/**
+ * Applies the zoom into the scale range.
+ * It changes the screen coordinates that the scale covers.
+ * Not the data that is displayed.
+ *
+ * @param scaleRange the original range in real screen coordinates.
+ * @param zoomRange the zoom range in percentage.
+ * @returns zoomed range in real screen coordinates.
+ */
+export const zoomScaleRange = (
+  scaleRange: [number, number] | number[],
+  zoomRange: [number, number] | number[],
+) => {
+  const rangeGap = scaleRange[1] - scaleRange[0];
+  const zoomGap = zoomRange[1] - zoomRange[0];
+
+  // If current zoom show the scale between p1 and p2 percents
+  // The range should be extended by adding [0, p1] and [p2, 100] segments
+  const min = scaleRange[0] - (zoomRange[0] * rangeGap) / zoomGap;
+  const max = scaleRange[1] + ((100 - zoomRange[1]) * rangeGap) / zoomGap;
+
+  return [min, max];
+};
+
+const zoomExtremums = (
+  extremums: [number, number] | number[],
+  zoomRange: [number, number],
+): [number, number] => {
+  const zoomGap = zoomRange[1] - zoomRange[0];
+  const extremumsGap = extremums[1] - extremums[0];
+
+  // const minZoomed = extremums[0] + extremumsGap * ((100 - zoomGap) / 100);
+  // const maxZoomed = extremums[1] - extremumsGap * ((100 - zoomGap) / 100);
+
+  const minZoomed = extremums[0] + (zoomRange[0] * extremumsGap) / zoomGap;
+  const maxZoomed = extremums[1] - ((100 - zoomRange[1]) * extremumsGap) / zoomGap;
+
+  // const minZoomed = extremums[0] * ((zoomRange[0] + 100) / 100);
+  // const maxZoomed = extremums[1] * (1 - (zoomRange[1] - zoomRange[0]) / 100);
+
+  return [minZoomed, maxZoomed];
+};
```

Comment:
@alexfauquette I can't seem to get this right. 😢 

As far as I understand, the `zoomScaleRange` above should "zoom out" the ranges, that is, the gap between the ranges increase in order to "zoom in" the data. 

But for `zoomExtremums` we need to "zoom in" the extremums in order to "filter" the data, so the gap between extremum values has to decrease, but not overlap each other. And I can't get that right.